src/_coustomware/socketMiddleware.js

```diff
@@ -0,0 +1,102 @@
+import { userActions } from '../_actions';
+import { config } from '../_helpers';
+import { userConstants } from '../_constants';
+
+const socketMiddleware = (function () {
+
+  var socket = null;
+
+  const onOpen = (ws, store, token) => evt => {
+    //Send a handshake, or authenticate with remote end
+    socket.send(token);
```

Comment:
This works because we only open the websocket connection after logging in, right?

Reply:
True.